File: color-shooter/src/core/src/com/salvai/centrum/input/CatchBackKeyProcessor.java
package com.salvai.centrum.input;

import com.badlogic.gdx.Input;
import com.badlogic.gdx.InputAdapter;
import com.badlogic.gdx.Screen;
import com.badlogic.gdx.scenes.scene2d.actions.Actions;
import com.salvai.centrum.CentrumGameClass;
import com.salvai.centrum.enums.GameType;
import com.salvai.centrum.screens.GameOverScreen;
import com.salvai.centrum.screens.LevelChooseScreen;
import com.salvai.centrum.screens.MenuScreen;
import com.salvai.centrum.utils.Constants;

public class CatchBackKeyProcessor extends InputAdapter {
    private CentrumGameClass game;
    private Screen screen;

    public CatchBackKeyProcessor(CentrumGameClass game, Screen screen) {
        this.game = game;
        this.screen = screen;
    }

    @Override
    public boolean keyDown(int keycode) {
        if (keycode == Input.Keys.BACK) {
            if (screen instanceof LevelChooseScreen) {
                ((LevelChooseScreen) screen).stage.addAction(Actions.sequence(Actions.fadeOut(Constants.FADE_TIME), Actions.run(new Runnable() {
                    @Override
                    public void run() {
                            game.setScreen(new MenuScreen(game));
                        screen.dispose();
                    }
                })));
            } else if (screen instanceof MenuScreen)
                return false;
            else if (screen instanceof GameOverScreen) {

                ((GameOverScreen) screen).stage.addAction(Actions.sequence(Actions.fadeOut(Constants.FADE_TIME), Actions.run(new Runnable() {
                    @Override
                    public void run() {
                        if (game.gameType == GameType.LEVEL)
                            game.setScreen(new LevelChooseScreen(game));
                        else
                            game.setScreen(new MenuScreen(game));
                        screen.dispose();
                    }
                })));

            }

        }
        return true;
    }
}

File: Color_Shooter/core/src/com/salvai/centrum/input/CatchBackKeyProcessor.kt
package com.salvai.centrum.input

import com.badlogic.gdx.Input
import com.badlogic.gdx.InputAdapter
import com.badlogic.gdx.Screen
import com.badlogic.gdx.scenes.scene2d.actions.Actions
import com.salvai.centrum.CentrumGameClass
import com.salvai.centrum.enums.GameType
import com.salvai.centrum.screens.GameOverScreen
import com.salvai.centrum.screens.LevelChooseScreen
import com.salvai.centrum.screens.MenuScreen
import com.salvai.centrum.utils.Constants

class CatchBackKeyProcessor(private val game: CentrumGameClass, private val screen: Screen?) : InputAdapter() {
    override fun keyDown(keycode: Int): Boolean {
        if (keycode == Input.Keys.BACK) {
            if (screen is LevelChooseScreen) {
                screen.stage.addAction(Actions.sequence(Actions.fadeOut(Constants.FADE_TIME), Actions.run(object : Runnable {
                    override fun run() {
                        game.setScreen(MenuScreen(game))
                        screen.dispose()
                    }
                })))
            } else if (screen is MenuScreen) return false
            else if (screen is GameOverScreen) {
                screen.stage.addAction(Actions.sequence(Actions.fadeOut(Constants.FADE_TIME), Actions.run(object : Runnable {
                    override fun run() {
                        if (game.gameType == GameType.LEVEL) game.setScreen(LevelChooseScreen(game))
                        else game.setScreen(MenuScreen(game))
                        screen.dispose()
                    }
                })))
            }
        }
        return true
    }
}
